src/util/chunk_fs.js

```diff
@@ -28,54 +28,50 @@ class ChunkFS extends stream.Transform {
         this._total_num_buffers = 0;
     }
 
-    _transform(chunk, encoding, callback) {
+    async _transform(chunk, encoding, callback) {
         try {
-            this._process_chunk(chunk, callback);
+            if (this.MD5Async) await this.MD5Async.update(chunk);
+            stats_collector.instance(this.rpc_client).update_namespace_write_stats({
```

Comment:
I would add this condition to help running this class it in tests that do not care about stats:
```suggestion
            if (this.rpc_client) {
                stats_collector.instance(this.rpc_client).update_namespace_write_stats({
```

Reply:
I'm using a dummy RPC for that matter, you want me to revert the dummy RCP code from the unit_tests?
```
const DUMMY_RPC = {
    object: {
        update_endpoint_stats: (...params) => null
    }
};
```